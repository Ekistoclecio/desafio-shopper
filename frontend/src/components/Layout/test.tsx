import { screen } from '@testing-library/react';
import { Layout } from '.';

import { renderWithTheme } from '@/utils/tests/helpers';

describe('<Layout />', () => {
  it('should render the Layout', () => {
    const { container } = renderWithTheme(<Layout />);

    const mainElement = screen.getByRole('main');
    expect(mainElement).toBeInTheDocument();

    const parentElement = mainElement.parentElement;
    if (parentElement) {
      const styles = getComputedStyle(parentElement);
      expect(styles.backgroundImage).toContain('url(');
    }

    expect(container.firstChild).toMatchSnapshot();
  });
});
