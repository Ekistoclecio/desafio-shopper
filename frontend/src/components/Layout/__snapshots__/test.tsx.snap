// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should render the Layout 1`] = `
<div
  class="MuiBox-root css-1t0qyeu"
>
  <main
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation5 css-1rt5ekp-MuiPaper-root"
    style="--Paper-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 5px 8px 0px rgba(0,0,0,0.14),0px 1px 14px 0px rgba(0,0,0,0.12);"
  />
</div>
`;
